prog/vnet/update_firewall_rules.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+class Prog::Vnet::UpdateFirewallRules < Prog::Base
+  subject_is :vm
+
+  label def update_firewall_rules
+    rules = vm.private_subnets.map(&:firewall_rules).flatten
+    allowed_ingress_ip4 = rules.select { !_1.ip6? && _1.accept? }.map { _1.ip.to_s }
+    blocked_ingress_ip4 = rules.select { !_1.ip6? && !_1.accept? }.map { _1.ip.to_s }
+    allowed_ingress_ip6 = rules.select { _1.ip6? && _1.accept? }.map { _1.ip.to_s }
+    blocked_ingress_ip6 = rules.select { _1.ip6? && !_1.accept? }.map { _1.ip.to_s }
+
+    vm.vm_host.sshable.cmd(<<ADD_RULES)
+sudo ip netns exec #{vm.inhost_name} bash -c 'sudo nft flush set inet fw_table blocked_ipv4_ips'
+sudo ip netns exec #{vm.inhost_name} bash -c 'sudo nft flush set inet fw_table blocked_ipv6_ips'
+sudo ip netns exec #{vm.inhost_name} bash -c 'sudo nft flush set inet fw_table allowed_ipv4_ips'
+sudo ip netns exec #{vm.inhost_name} bash -c 'sudo nft flush set inet fw_table allowed_ipv6_ips'
+sudo ip netns exec #{vm.inhost_name} bash -c 'sudo nft add element inet fw_table blocked_ipv4_ips { #{blocked_ingress_ip4.join(",")} }'
+sudo ip netns exec #{vm.inhost_name} bash -c 'sudo nft add element inet fw_table blocked_ipv6_ips { #{blocked_ingress_ip6.join(",")} }'
+sudo ip netns exec #{vm.inhost_name} bash -c 'sudo nft add element inet fw_table allowed_ipv4_ips { #{allowed_ingress_ip4.join(",")} }'
+sudo ip netns exec #{vm.inhost_name} bash -c 'sudo nft add element inet fw_table allowed_ipv6_ips { #{allowed_ingress_ip6.join(",")} }'
+ADD_RULES
```

Comment:
Any reason to do this in so many commands? Each one forks several processes (sudo, ip, bash, nft, sudo again). How about using stdin with the `nft --file -` formation, so you can go through that rigamarole just once?

Reply:
I have updated but we cannot get away from 2 separate commands. The reason is, even if we give a new set definition with empty element set, nftables doesn't simply update the sets when there are missing elements. It works fine for adding but not for removing. Therefore, we have to flush first and then redefine the set. That's what I do now.